### ISSUE TITLE
Unexpected AuthError from _check_event_auth

### ISSUE THREAD

**Issue by richvdh** [matrix-org/synapse#11513]
#11011 added a check, boldly asserting that some code was unreachable. The evidence suggests otherwise. We should find out why this is happening.

**Comment by richvdh** [comment]
Example stacktrace:

```
AuthError: 403: Auth events could not be found
  File "synapse/handlers/federation_event.py", line 983, in _process_received_pdu
    context,
  File "synapse/handlers/federation_event.py", line 1371, in _check_event_auth
    origin, event
  File "synapse/handlers/federation_event.py", line 1683, in _load_or_fetch_auth_events_for_event
    raise AuthError(code=HTTPStatus.FORBIDDEN, msg="Auth events could not be found")
```


https://sentry.matrix.org/sentry/synapse-matrixorg/issues/233645
https://sentry.matrix.org/sentry/synapse-modular/issues/235420

**Comment by cremesk** [comment]
Any idea howto fix this issue?

I see the same if i try to join #mobian:matrix.org

![image](https://user-images.githubusercontent.com/7526990/162187865-5ed10247-0b78-467c-8449-73622e79a600.png)

also this:

```
2022-04-07 11:21:47,907 - synapse.http.server - 690 - WARNING - POST-8517 - Not sending response to request <SynapseRequest at 0x7f92829b01c0 method='POST' uri='/_synapse/replication/remote_join/%21ThIzZNArdoSkaWndMq%3Amatrix.org/%40creme%3Aenvs.net/CxikfBNBzQ' clientproto='HTTP/1.1' site='9093'>, already disconnected.
```

i create also an local alias on an other server ( #mobian:tchncs.de ) and try to join via this alias. same problem :/

**Comment by richvdh** [comment]
#11001 made `AuthError` a documented exception from `_check_event_auth`, so the problem is now simply that we log it. Fixed by #13145.
